### PR TITLE
Add kafka consumer

### PR DESCRIPTION
**Pull request by evq**

### CODE REVIEW

File: bat-utils/lib/runtime-kafka.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    this.runtime = runtime
+    this.config = kafka
+    this.topicHandlers = {}
+  }
+  async connect () {
+    // FIXME testing only
+    const partitionCount = 1
+
+    this.producer = new NProducer(kafkaConfiguration, null, partitionCount)
+    this.producer.on('error', error => console.error(error))
```
Comment (maikelmclauflin) [inline bat-utils/lib/runtime-kafka.js]: let's capture this in sentry too

File: test/eyeshade/suggestions.integration.test.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  await runtime.kafka.consume()
+
+  await producer.send('grant-suggestions', suggestionType.toBuffer(example))
+
+  await messagesPromise
+
+  const { body } = await eyeshadeAgent.get(`/v1/accounts/${encodeURIComponent(channel)}?pending=true`)
+  console.log(body)
+
+  t.is(1, 1)
```
Comment (maikelmclauflin) [inline test/eyeshade/suggestions.integration.test.js]: let's just use t.plan(0) or test the body for being object

File: test/eyeshade/suggestions.integration.test.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    {
+      'type': 'ugp',
+      'amount': '10',
+      'cohort': 'control',
+      'promotion': '6820f6a4-c6ef-481d-879c-d2c30c8928c3'
+    }
+  ]
+}
+
+const suggestionType = avro.parse({
```
Comment (maikelmclauflin) [inline test/eyeshade/suggestions.integration.test.js]: can we export this so we don't have it twice in code

File: package.json [hunk 4]
```diff
@@ -75,6 +76,7 @@
     "@sentry/node": "^5.7.1",
     "@uphold/uphold-sdk-javascript": "github:brave-intl/uphold-sdk-javascript#bump-lodash-merge",
     "accept-language-parser": "1.4.1",
+    "avro-js": "1.9.1",
```
Comment (maikelmclauflin) [inline package.json]: tried npm i locally and did not have any luck. is there a specific node version that should be used i tried 8.9.2 and 10.15.3

File: test/eyeshade/suggestions.integration.test.js (hunk 2 above)
Comment (evq) [reply test/eyeshade/suggestions.integration.test.js]: yeah this test is wip

File: test/eyeshade/suggestions.integration.test.js (hunk 3 above)
Comment (evq) [reply test/eyeshade/suggestions.integration.test.js]: wip

File: package.json (hunk 4 above)
Comment (evq) [reply package.json]: whaaa? there should not be any build issues with this, it's native javascript. did you have problems with `sinek` maybe?

Comment (maikelmclauflin) [reply package.json]: sorry. yeah. could be. just npm i failed. not sure which part

File: eyeshade/workers/suggestions.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+              surveyorId
+            ])
+          }
+        }
+      } catch (e) {
+        await client.query('ROLLBACK')
+        runtime.captureException(e, { extra: { topic: 'grant-suggestions' } })
+        throw e
+      }
+      await client.query('COMMIT')
```
Comment (xtat) [inline eyeshade/workers/suggestions.js]: suddenly relieved that I am not the only one using raw SQL to do financial transactions :ledger:

File: eyeshade/workers/suggestions.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  runtime.kafka.on('grant-suggestions', async (messages) => {
+    const client = await runtime.postgres.connect()
+    try {
+      await client.query('BEGIN')
+      try {
+        for (let message of messages) {
+          const buf = Buffer.from(message.value, 'binary')
+          const suggestion = suggestionType.fromBuffer(buf)
+
+          console.log(suggestion)
```
Comment (maikelmclauflin) [inline eyeshade/workers/suggestions.js]: probably ok to remove

File: eyeshade/lib/suggestions.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
+const avro = require('avro-js')
+
+module.exports.suggestionType = avro.parse({
```
Comment (maikelmclauflin) [inline eyeshade/lib/suggestions.js]: was this not going to use the schema module?

**Pull request merged**